plugins/filter/consolidate.py

```diff
@@ -0,0 +1,1249 @@
+#
+# -*- coding: utf-8 -*-
+# Copyright 2022 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+#
+
+"""
+The consolidate filter plugin
+"""
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+    name: consolidate
+    author: Sagar Paul (@KB-perByte)
+    version_added: "2.6.0"
+    short_description: Consolidate facts together on common attributes.
+    description:
+        - This plugin presents collective structured data including all supplied facts grouping on common attributes mentioned.
+        - All other boolean parameter defaults to False unless parameters is explicitly mentioned.
+        - Using the parameters below- C(data_source|ansible.utils.consolidate(fail_missing_match_key=False)))
+    options:
+      data_source:
+        description:
+        - This option represents a list of dictionaries to perform the operation on.
+        - For example C(facts_source|ansible.utils.consolidate(fail_missing_match_key=False))), in this case C(facts_source) represents this option.
+        type: list
+        elements: dict
+        required: True
+        suboptions:
+          data:
+            description: Specify facts data that gets consolidated.
+            type: raw
+            required: True
+          match_key:
+            description: Specify key to match on.
+            type: str
+            required: True
+          prefix:
+            description: Specify the prefix with which the result set be created.
+            type: str
+            required: True
+      fail_missing_match_key:
+        description: Fail if match_key is not found in a specific data set.
+        type: bool
+      fail_missing_match_value:
+        description: Fail if a keys to match in not same accross all data sets.
+        type: bool
+      fail_duplicate:
+        description: Fail if duplicate values for any key is found.
+        type: bool
+"""
+
+EXAMPLES = r"""
```

Comment:
Can we start with a super simple example?  maybe with a small a b c data set that shows how it works before showing a large data set? if you look at get_path or update_fact, they go from simple to complex in the examples.  We could end with an example using resource modules to get the data, or just the facts module to get different resources

Reply:
Showing the behavior of the different failure option might e easier with a small sample data set as well